lib/oxrdf/src/triple.rs

```diff
@@ -738,6 +751,26 @@ impl Triple {
         }
     }
 
+    /// Builds an RDF [triple](https://www.w3.org/TR/rdf11-concepts/#dfn-rdf-triple).
+    #[inline]
+    pub fn new_maybe(
+        subject: impl Into<Term>,
+        predicate: impl Into<Term>,
+        object: impl Into<Term>,
+    ) -> Option<Self> {
+        match (
+            Into::<Option<Subject>>::into(Into::<Term>::into(subject)),
+            Into::<Option<NamedNode>>::into(Into::<Term>::into(predicate)), 
+            Into::<Option<Term>>::into(Into::<Term>::into(object))) {
+            (Some(subject), Some(predicate), Some(object)) => Some(Self {
+                subject,
+                predicate,
+                object,
+            }),
+            _ => None,
+        }
+    }
```

Comment:
I see the point of this function but I would make it return a `Result<Self, TermCastError>` to highlight that it returns an error with `TermCastError` probably the same error as the one in the new `TryFrom` implementations.